src/go/k8s/pkg/resources/pdb_test.go

```diff
@@ -79,6 +80,12 @@ func TestEnsure_PDB(t *testing.T) {
 			err = c.Get(context.Background(), pdb.Key(), actual)
 			assert.NoError(t, err, tt.name)
 			assert.Equal(t, tt.expectedObject.Spec, actual.Spec)
+
+			// retrieved object and new object should produce no diff and thus
+			// not be updated
+			updated, err := resources.Update(context.TODO(), actual, actual, c, ctrl.Log.WithName("test"))
+			assert.NoError(t, err)
+			assert.False(t, updated)
```

Comment:
this test was failing before I bumped the objectmatcher lib and started ignoring the selector